website/addons/osfstorage/views.py

```diff
@@ -217,6 +222,10 @@ def osfstorage_delete(file_node, payload, node_addon, **kwargs):
 
     except exceptions.FileNodeCheckedOutError:
         raise HTTPError(httplib.FORBIDDEN)
+    except exceptions.FileNodeIsPrimaryFile:
+        raise HTTPError(httplib.FORBIDDEN, data={
```

Comment:
This should be METHOD_NOT_ALLOWED, because the DELETE method is not allowed for the file in this instance.